src/main/webapp/components/navbar.vue

```diff
@@ -0,0 +1,158 @@
+<template>
+  <nav class="blue nav-bar">
+    <a href="index.html" class="bold" id="title">starfish</a>
+    <img src="../assets/starfish.png" id="logo">
+    <div id="action-items">
+      <a href="#" id="post-action">PostANote</a>
+      <div class="dropdown">
+        <button class="dropbtn" @click.stop="toggleDropdown">
+          <img id="user-profile" src="../assets/user.svg">
+        </button>
+        <div class="dropdown-content" v-if="showDropdown" v-click-outside="hideDropdown">
+          <img src="../assets/user.svg" height="115px"/>
+          <p class="bold" id="dropdown-name">{{ user.name }}</p>
+          <p id="dropdown-points">{{ user.points }} points</p>
+          <a class="dropdown-link" href="#">My Profile</a> 
+          <a class="dropdown-link" href="#">Favorite Notes</a>
+          <a class="dropdown-link" href="#">Logout</a>
+        </div>
+      </div>
+    </div> 
+  </nav>
+</template>
+
+<script>
+Vue.directive('click-outside', {
+  bind: function (el, binding, vnode) {
+    this.event = function(event) {
+      if(!(el == event.target || el.contains(event.target))) {
+        vnode.context[binding.expression](event);
+      }
+    };
+    document.body.addEventListener('click', this.event);
+  },
+  unbind: function(el) {
+    document.body.removeEventListener('click', this.event);
+  }
+});
+
+module.exports = {
+  data: function() {
+    return {
+      showDropdown: false
+    }
+  },
+  props: {
+    user: Object
+  },
+  methods: {
+    hideDropdown() {
+      this.showDropdown = false;
+    },
+    toggleDropdown() {
+      this.showDropdown = !this.showDropdown;
+    }
+  }
+}
+</script>
+
+<style>
+.bold {
+  font-weight: bold;
+}
+
+.blue {
+  background-color: #004aad;
+}
+
+.nav-bar {
+  align-items: center;
+  display: flex;
+  font-family: 'Roboto';
+  height: 80px;
+}
+
+.nav-bar a {
+  float: left;
+  font-size: 1.75em;
+  color: white;
+  text-decoration: none;
+}
+
+#title {
+  font-size: 2.75em;
+  margin: 0 10px 0 20px;
+}
+
+#logo {
+  height: 60%;
+  padding: 0 0 5px 0;
+}
+
+#action-items {
+  height: 100%;
+  margin-left: auto;
+}
+
+#post-action {
+  height: 80px;
+  line-height: 80px;
+  padding: 0;
+}
+
+.dropdown {
+  overflow: hidden;
+  padding: 0 10px 0 20px;
+}
+
+.dropdown .dropbtn {
+  background-color: inherit;
+  border: none;
+  color: white;
+  font-size: 16px;
+  font-family: inherit;
+  height: 100%;
+  margin: 0;
+  outline: none;
+}
+
+.dropdown .dropbtn:hover {
+  cursor: pointer;
+}
+
+.dropbtn img {
+  height: 65%;
+}
+
+.dropdown-content {
+  align-items: center;
+  background-color: #f9f9f9;
```

Comment:
You have a separate class for the background-color blue above so for consistency it might make sense to make separate classes for this color and in the other selector below. Not too big of a deal though.